astropy/utils/abcs.py

```diff
@@ -0,0 +1,277 @@
+# -*- coding: utf-8 -*-
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Astropy's ABCs."""
```

Comment:
I haven't been following this closing. Why is this needed?

Reply:
``utils/collections`` and ``utils/codegen`` aren't the most natural places to put this. Just trying to find a home for this cross-package base class.

Reply:
Also, I want to push an ``InstanceDescriptor`` abc so we can unify of all our instance-level descriptors, e.g. in ``UnifiedReadWrite`` and ``DataInfo``.